src/Packetery/Module/Options/Page.php

```diff
@@ -484,27 +484,52 @@ public function create_form(): Form {
+			[
+				OptionsProvider::DIMENSIONS_UNIT_CM => 'cm',
+				OptionsProvider::DEFAULT_DIMENSIONS_UNIT_MM => 'mm',
+			]
+		)
+		->setDefaultValue( $this->optionsProvider::DEFAULT_DIMENSIONS_UNIT_MM );
+
 		$container->addCheckbox( 'default_dimensions_enabled', __( 'Enable default dimensions', 'packeta' ) )
 			->addCondition( Form::EQUAL, true )
 				->toggle( '#packetery-default-dimensions-value' );
 
-		$container->addText( 'default_length', __( 'Length', 'packeta' ) . ' (mm)' )
+		$container->addText( 'default_length', __( 'Length', 'packeta' ) )
 			->addConditionOn( $form[ self::FORM_FIELDS_CONTAINER ]['default_dimensions_enabled'], Form::EQUAL, true )
 				->setRequired()
-				->addRule( Form::INTEGER )
-				->addRule( Form::MIN, null, 0 );
-
-		$container->addText( 'default_height', __( 'Height', 'packeta' ) . ' (mm)' )
+			->addConditionOn( $form[ self::FORM_FIELDS_CONTAINER ]['dimensions_unit'], Form::EQUAL, $this->optionsProvider::DEFAULT_DIMENSIONS_UNIT_MM )
+				->addRule( Form::INTEGER, __( 'Provide a full number!', 'packeta' ) )
+				->addRule( Form::MIN, 'Value must be greater than 0', 1 )
+			->elseCondition()
+				->addRule( Form::FLOAT, __( 'Provide a decimal value!', 'packeta' ) )
+				->addRule( Form::MIN, 'Value must be greater than 0', 0.1 )
+			->endCondition();
+
+		$container->addText( 'default_height', __( 'Height', 'packeta' ) )
 			->addConditionOn( $form[ self::FORM_FIELDS_CONTAINER ]['default_dimensions_enabled'], Form::EQUAL, true )
 				->setRequired()
-				->addRule( Form::INTEGER )
-				->addRule( Form::MIN, null, 0 );
-
-		$container->addText( 'default_width', __( 'Width', 'packeta' ) . ' (mm)' )
+			->addConditionOn( $form[ self::FORM_FIELDS_CONTAINER ]['dimensions_unit'], Form::EQUAL, $this->optionsProvider::DEFAULT_DIMENSIONS_UNIT_MM )
+				->addRule( Form::INTEGER, __( 'Provide a full number!', 'packeta' ) )
+				->addRule( Form::MIN, 'Value must be greater than 0', 1 )
+			->elseCondition()
+				->addRule( Form::FLOAT, __( 'Provide a decimal value!', 'packeta' ) )
+				->addRule( Form::MIN, 'Value must be greater than 0', 0.1 )
+			->endCondition();
+
+		$container->addText( 'default_width', __( 'Width', 'packeta' ) )
 			->addConditionOn( $form[ self::FORM_FIELDS_CONTAINER ]['default_dimensions_enabled'], Form::EQUAL, true )
-				->setRequired()
-				->addRule( Form::INTEGER )
-				->addRule( Form::MIN, null, 0 );
+				->setRequired( 'Width is required when default dimensions are enabled.' )
+			->addConditionOn( $form[ self::FORM_FIELDS_CONTAINER ]['dimensions_unit'], Form::EQUAL, $this->optionsProvider::DEFAULT_DIMENSIONS_UNIT_MM )
+				->addRule( Form::INTEGER, __( 'Provide a full number!', 'packeta' ) )
+				->addRule( Form::MIN, 'Value must be greater than 0', 1 )
+			->elseCondition()
+				->addRule( Form::FLOAT, __( 'Provide a decimal value!', 'packeta' ) )
+				->addRule( Form::MIN, 'Value must be greater than 0', 0.1 )
+			->endCondition();
 
 		// TODO: Packet status sync.
```

Comment:
Oproti #580 chybí `foreach ( [ 'default_length', 'default_width', 'default_height' ]` - je to správně?

Reply:
Po analyze jsem zjistil, ze toto tam skutecne chybi. Pridano zpet a upravena logika v loopu.